File: 961. N-Repeated Element in Size 2N Array.py
class Solution(object):
    def repeatedNTimes(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        t = {}
        for num in nums:
            if num in t:
                t[num] += 1
            else:
                t[num] = 1
                
        n = len(nums) / 2
        for key in t:
            if t[key] == n:
                return key

File: 961. N-Repeated Element in Size 2N Array.java
import java.util.*;
class Solution {
    public int repeatedNTimes(int[] nums) {
        Hashtable<Integer, Integer> t = new Hashtable<>();
        // go over each num 
        for (int num : nums) {
            // if num in t, increase count by 1
            if (t.containsKey(num)) {
                return num;
            // if num not in t, add it with count = 1
            } else {
                t.put(num, 1);
            }
        }
        // this line should not be executed
        return -1;
    }
}